src/riakc_ts.erl

```diff
@@ -69,20 +71,20 @@ query(Pid, Query, Interpolations, Cover) when is_binary(Cover) ->
 query_common(Pid, Query, Interpolations, Cover)
   when is_pid(Pid), is_list(Query) ->
     Message = riakc_ts_query_operator:serialize(
-                iolist_to_binary(Query), Interpolations),
+                convert_to_binary(Query), Interpolations),
     Response = server_call(Pid, Message#tsqueryreq{cover_context = Cover}),
     riakc_ts_query_operator:deserialize(Response).
 
 
--spec get_coverage(pid(), table_name(), Query::string()) ->
-                          {ok, Entries::[term()]} | {error, term()}.
 %% @doc Generate a parallel coverage plan for the specified query
-get_coverage(Pid, Table, Query)
-  when is_pid(Pid), (is_binary(Table) orelse is_list(Table)), is_list(Query) ->
+-spec get_coverage(Pid::pid(),
+                   Table::binary(),
+                   QueryText::binary()) -> {ok, Entries::[term()]} | {error, term()} | {'EXIT', any()}.
+get_coverage(Pid, Table, QueryText) ->
     server_call(Pid,
```

Comment:
The Entries happen to be returned to the client raw. This means the caller of `riakc_ts:get_coverage` needs to do the matching on `#tscoverageentry{}` fields, while the record itself is internal to riak_pb. (Leaving this note here as a todo item, even though it goes outside of the scope of this PR).